File: src/main/java/com/back/nbe9112team06/global/security/BeanConfig.java
package com.back.nbe9112team06.global.security;

import org.springframework.boot.jackson.autoconfigure.JsonMapperBuilderCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.ProblemDetail;
import org.springframework.http.converter.json.ProblemDetailJacksonMixin;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

@Configuration
public class BeanConfig {
    @Bean
    public PasswordEncoder passwordEncoder() {
        return new BCryptPasswordEncoder();
    }

    @Bean
    public JsonMapperBuilderCustomizer problemDetailMixinCustomizer() {
        return builder -> {
            //  ProblemDetail 에 Mixin 적용 (properties 평탄화)
            builder.addMixIn(ProblemDetail.class, ProblemDetailJacksonMixin.class);
        };
    }
}

File: src/main/kotlin/com/back/nbe9112team06/global/security/BeanConfig.kt
package com.back.nbe9112team06.global.security

import org.springframework.boot.jackson.autoconfigure.JsonMapperBuilderCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.ProblemDetail
import org.springframework.http.converter.json.ProblemDetailJacksonMixin
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder
import tools.jackson.databind.json.JsonMapper

@Configuration
class BeanConfig {
    @Bean
    fun passwordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }

    @Bean
    fun problemDetailMixinCustomizer(): JsonMapperBuilderCustomizer {
        return JsonMapperBuilderCustomizer { builder: JsonMapper.Builder ->
            //  ProblemDetail 에 Mixin 적용 (properties 평탄화)
            builder.addMixIn(ProblemDetail::class.java, ProblemDetailJacksonMixin::class.java)
        }
    }
}
